runtime/lua/vim/lsp/util.lua

```diff
@@ -273,12 +273,20 @@ end
 -- Returns text that should be inserted when selecting completion item. The precedence is as follows:
 -- textEdit.newText > insertText > label
 -- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
-local function get_completion_word(item)
+local function get_completion_word(item, prefix)
   if item.textEdit ~= nil and item.textEdit.newText ~= nil then
+    local start_range = item.textEdit.range["start"]
+    local end_range = item.textEdit.range["end"]
+    local newText = ""
+    if start_range.line == end_range.line and start_range.character == end_range.character then
+      newText = prefix .. item.textEdit.newText
+    else
+      newText = item.textEdit.newText
```

Comment:
init newText with `item.textEdit.newText` and you won't need the `else`.

Reply:
Nice catch! Resolved in 5675715.